core/services/pipeline/task.divide_test.go

```diff
@@ -193,3 +196,38 @@ func TestDivideTask_Overflow(t *testing.T) {
 	assert.False(t, runInfo.IsRetryable)
 	require.Equal(t, pipeline.ErrDivisionOverlow, errors.Cause(result.Error))
 }
+
+func TestDivide_Example(t *testing.T) {
+	t.Parallel()
+
+	dag := `
+ds1 [type=memo value=10000.1234]
+
+ds2 [type=memo value=100]
+
+div_by_ds2 [type=divide divisor="$(ds2)"]
+
+multiply [type=multiply times=10000 index=0]
+
+ds1->div_by_ds2->multiply;
+`
+
+	db := pgtest.NewSqlxDB(t)
+	cfg := configtest.NewTestGeneralConfig(t)
+	btORM := bridgesMocks.NewORM(t)
+	r, _ := newRunner(t, db, btORM, cfg)
+
+	spec := pipeline.Spec{DotDagSource: dag}
+	vars := pipeline.NewVarsFrom(nil)
+
+	lggr := logger.TestLogger(t)
+	_, trrs, err := r.ExecuteRun(testutils.Context(t), spec, vars, lggr)
+	require.NoError(t, err)
+
+	require.Len(t, trrs, 4)
+
+	finalResult := trrs[3]
+
+	assert.Nil(t, finalResult.Result.Error)
+	assert.Equal(t, "1000012.34", finalResult.Result.Value.(decimal.Decimal).String())
```

Comment:
/nit if it bothers you, i might hardcode `10000.1234` into a test var and then do the actual math here at the end